### PR TITLE
make burgov/key-value optional

### PR DESCRIPTION
**Pull request by dbu**

### CODE REVIEW

**Comment by ElectricMaxxx** [comment]
mh, we had that discussion when introducing that exception in our extension class. We said just not displaying the form is a silent error no one would recognize, why is it different now?

**Comment by dbu** [comment]
making it a hard dependency seems just wrong to me. in menu admin, we have a special configuration that is off by default that needs it. we only throw the exception if that config is on. we could do something similar here. and as seo is already relying on the form type, we would get rid of the problem completely in the sonata integration.

**Comment by dbu** [comment]
okay, changed this a bit. if you agree with the new logic, i will adjust the tests as well. they currently expect the old behaviour

**Comment by dbu** [comment]
ping @ElectricMaxxx what do you think about the code? if you agree i will adjust the tests to the new format and update the changelog.

File: src/DependencyInjection/CmfSeoExtension.php [hunk 1]
```diff
@@ -52,7 +52,7 @@ public function load(array $configs, ContainerBuilder $container)
         $configuration = new Configuration();
         $config = $this->processConfiguration($configuration, $configs);
 
-        $loader = new XmlFileLoader($container, new FileLocator(__DIR__.'/../Resources/config'));
+        $loader = new XmlFileLoader($container, new FileLocator(__DIR__ . '/../Resources/config'));
```
Comment (ElectricMaxxx) [inline src/DependencyInjection/CmfSeoExtension.php]: I think that current style is without any spaces next to `.`

File: src/DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -293,8 +293,18 @@ private function addFormSection($nodeBuilder)
                             ->scalarNode('seo_metadata')->defaultNull()->end()
                         ->end()
                     ->end()
+                    ->arrayNode('options')
+                        ->addDefaultsIfNotSet()
+                        ->children()
+                            ->enumNode('advanced')
```
Comment (ElectricMaxxx) [inline src/DependencyInjection/Configuration.php]: With the info it sounds ok, but the isn't really speaking else.
What we do with that key-value stuff is arbitrary metadata, why not going in that direction?

File: src/Resources/config/services.xml [hunk 3]
```diff
@@ -15,7 +15,7 @@
 
         <service id="cmf_seo.form.type.seo_metadata" class="Symfony\Cmf\Bundle\SeoBundle\Form\Type\SeoMetadataType">
             <argument>%cmf_seo.form.data_class.seo_metadata%</argument>
-            <argument>%cmf_seo.form_mode_orm%</argument>
+            <argument>%cmf_seo.form.options%</argument>
             <tag name="form.type" alias="seo_metadata" />
```
Comment (ElectricMaxxx) [inline src/Resources/config/services.xml]: We don't need that tag anymore right?

**Comment by ElectricMaxxx** [comment]
Activated StyleCI and fixed against master, can you rebase please?

File: src/DependencyInjection/CmfSeoExtension.php (hunk 1 above)
Comment (dbu) [inline src/DependencyInjection/CmfSeoExtension.php]: hm, did not even change that actively. i wonder if phpstorm did that on save or what :-|

File: src/DependencyInjection/Configuration.php (hunk 2 above)
Comment (dbu) [inline src/DependencyInjection/Configuration.php]: you are very right. going for `generic_metadata`.

File: src/Resources/config/services.xml (hunk 3 above)
Comment (dbu) [inline src/Resources/config/services.xml]: if we would not, we could not use a constructor afaik. see http://symfony.com/doc/current/form/create_custom_field_type.html#creating-your-field-type-as-a-service

**Comment by dbu** [comment]
rebased and fixed style issues. this is now ready to merge if the tests agree.

**Comment by ElectricMaxxx** [comment]
@dbu can you please add some ExtensionTests for setting storage and options and ad it to the scheme file (with tests too)

File: src/Resources/config/services.xml (hunk 3 above)
Comment (wouterj) [inline src/Resources/config/services.xml]: Yes, tag is not deprecated. The `alias` attribute however can be removed.

Comment (ElectricMaxxx) [inline src/Resources/config/services.xml]: Sure, i ment the alias and not the tag :-)

**Comment by ElectricMaxxx** [comment]
@dbu can you do this one as second on your friday?

**Comment by dbu** [comment]
fixed the existing tests, added some more and then fixed the stuff the tests reveiled. hope i got it right with the fixxmlconfig?

**Pull request merged**